### PR TITLE
Generalize viewer slices logic

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request further generalizes the viewer slices logic (used for the slice plugin) to be less spectral-axis specific.  This is not expected to have any behavior consequences in jdaviz, but fixes support for slicing in lcviz (and other future downstream uses).

Closes #2253

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by kecnry** [comment]
it would be nice to get this in a bugfix so that lcviz can update to use jdaviz 3.10, but if anyone thinks this doesn't belong in a bugfix release, we can hold it back until 3.11/4.0 if necessary.

**Comment by rosteen** [comment]
I'm fine with it going in a bugfix release.

**Review by rosteen — APPROVED** [review]
I tested in Cubeviz and everything still seems to work. My minor commentary isn't a blocker.

File: jdaviz/configs/cubeviz/plugins/viewers.py [hunk 1]
```diff
@@ -109,23 +109,22 @@ def slice_values(self):
...
-                data_obj = layer.layer.data.get_component(world_comp_ids[self.slice_index]).data
-                data_units = layer.layer.data.get_component(world_comp_ids[self.slice_index]).units
+                data_comp = layer.layer.data.get_component(world_comp_ids[self.slice_index])
             except (AttributeError, KeyError):
                 continue
 
-            # Find the spectral axis
-            data_spec_axis = np.asarray(data_obj.take(0, take_inds[0]).take(0, take_inds[1]),  # noqa
-                                        dtype=float)
+            data = np.asarray(data_comp.data.take(0, take_inds[0]).take(0, take_inds[1]),  # noqa
```
[the remark this answers is not in the document]
Comment (kecnry) [reply jdaviz/configs/cubeviz/plugins/viewers.py:116]: there is a property higher up where the viewer can set which axis is the slice axis (`slice_index`) which then controls the value of `take_inds` to take the remaining axes and slice only on the slice axis.  So _hopefully_ any changes in the future would just be to that property which is currently set to the last axis (`slice_index = 2`) 🤞

[the remark this answers is not in the document]
Comment (rosteen) [reply jdaviz/configs/cubeviz/plugins/viewers.py:116]: Great!

**Review by bmorris3 — APPROVED** [review]
This looks good. We'll need this for the ramp viewer eventually too, happy to see it wasn't too much trouble!

**Pull request merged**